### PR TITLE
feat: support mesc

### LINKED ISSUE #8502: feat(`config`): add MESC support
### Component

Forge

### Describe the feature you would like

https://github.com/paradigmxyz/mesc

I'm still trying to wrap my head around this tool, but I'm trying to see how/where/why this would fit into foundry... or even if it makes sense. Seems like it would be a bit annoying to have _both_ a `foundry.toml` and a mesc file? 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by onbjerg**
## Motivation

mesc is a cross-tool endpoint management tool that allows users to manage endpoints across chains, set defaults, and aliases.

We already have some pre-existing ways to do this in Foundry (i.e. rpc aliases in `foundry.toml`), but supporting mesc makes sense going forward. I'd also recommend that we consider deprecating the endpoint management stuff we have currently at some point in the future.

## Solution

Before looking at rpc aliases in `foundry.toml`, we query mesc first. mesc can handle things like the chain name, chain ID, a raw url, aliases and more.

This PR does not change the default foundry url based on mesc as I felt that might be too big of a change, given setting an RPC url in some contexts ends up enabling forking or other behavior

Closes https://github.com/foundry-rs/foundry/issues/8502

### CODE REVIEW

File: crates/config/src/lib.rs [hunk 1]
```diff
@@ -1092,6 +1099,10 @@ impl Config {
         &self,
         maybe_alias: &str,
     ) -> Option<Result<Cow<'_, str>, UnresolvedEnvVarError>> {
+        if let Ok(Some(endpoint)) = mesc::get_endpoint_by_query(maybe_alias, Some("foundry")) {
```
Comment (klkvr) [inline crates/config/src/lib.rs:1102]: this will prefer mesc aliases over ones in foundry.toml, should we flip it to make it easier to override?

**Review by klkvr — APPROVED** [review]
lgtm, just a question on priority of aliases

File: crates/config/src/lib.rs (hunk 1 above)
Comment (onbjerg) [reply crates/config/src/lib.rs:1109]: changed the order; foundry alias first, mesc second

**Review by mattsse — APPROVED** [review]
cool, love this

File: crates/config/src/lib.rs [hunk 2]
```diff
@@ -1093,7 +1102,15 @@ impl Config {
         maybe_alias: &str,
     ) -> Option<Result<Cow<'_, str>, UnresolvedEnvVarError>> {
         let mut endpoints = self.rpc_endpoints.clone().resolved();
-        Some(endpoints.remove(maybe_alias)?.map(Cow::Owned))
+        if let Some(endpoint) = endpoints.remove(maybe_alias) {
+            return Some(endpoint.map(Cow::Owned))
+        }
+
+        if let Ok(Some(endpoint)) = mesc::get_endpoint_by_query(maybe_alias, Some("foundry")) {
```
Comment (DaniPopes) [inline crates/config/src/lib.rs:1109]: We're ignoring the Err variant here, is this right? @onbjerg @klkvr

Comment (onbjerg) [reply crates/config/src/lib.rs:1109]: there are errors in there related to eg mesc not being enabled and such, which is why i chose to ignore it. don't want it to fail for people not using mesc

**Review by Nithin200487 — APPROVED** [review]

**Pull request merged**